packages/wpcom.js/src/lib/util/request.js

```diff
@@ -29,15 +29,24 @@ Req.prototype.get = function ( params, query, fn ) {
  * Make `update` request
  * @param {Object | string} params
  * @param {Object} [query] - query object parameter
- * @param {Object} body - body object parameter
+ * @param {Object} [body] - body object parameter
  * @param {Function} fn - callback function
  */
 Req.prototype.post = Req.prototype.put = function ( params, query, body, fn ) {
 	if ( undefined === fn ) {
 		if ( undefined === body ) {
-			body = query;
-			query = {};
+			if ( 'function' === typeof query ) {
+				// post( params, fn )
+				fn = query;
+				body = null;
+				query = {};
+			} else {
+				// post( params, body )
+				body = query;
+				query = {};
+			}
```

Comment:
Unit tests discovered that this fix is need when `sendRequest` no longer does the defaulting. If you call:
```js
post( params, fn );
```
then `sendRequest` was called like:
```js
sendRequest( params, {}, fn, undefined );
```
i.e., the `body` param is missing and the `fn` param is shifted to the left. After adding a check whether `query` is a function the same call will get the correct form without missing arguments:
```js
sendRequest( params, {}, null, fn );
```

Reply:
The defaulting of the missing params is a hard-to-understand quirk: you pass 3 params instead of 4 and it's interpreted as if one of the parameters in the middle was missing, and everything is shifted to the left, `body` (param 3) is interpreted as `fn` (param 4) etc. If we were designing this API from scratch today, I'm pretty sure we wouldn't do this again. 🙂

Therefore, I'd rather resolve this quirk as soon as possible, right in the `get`/`post` method, and always send only "normalized" params to the internal methods. Also, the `post` and `get` methods are a bit different,